torch/distributed/pipeline/sync/_balance/profile.py

```diff
@@ -91,9 +96,12 @@ def profile_sizes(
     batch = Batch(input)
     sizes: List[int] = []
 
-    latent_scale = batch[0].size(0) / chunks
+    tensor_idx = batch.find_tensor_idx()
+    latent_scale = batch[tensor_idx].size(0) // chunks
```

Comment:
I am not familiar enough with this code, so I just wanted to confirm you want the truncation behavior with this integer division.

Reply:
There was a mistake on my part and I corrected it. (Like the original, set the micro-batch size to 1, pass it through the pipeline, and return to the actual micro-batch size to be used.) Thank you very much for confirming again. However, it seems that `find_tensor_idx` should still be used because the order of tensors in a batch cannot be guaranteed.